### DISCUSSION TITLE
How can I set up CI/CD with Dagster using a bitbucket repo?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17524] [Q&A]
Hello, how can I set up CI/CD with Dagster using a bitbucket repo?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C02LJ7G0LAZ/p1698770585917639?thread_ts=1698770585.917639&cid=C02LJ7G0LAZ).*

**Comment by shalabhc** [comment]
To set up CI/CD using bitbucket, you can follow instructions here for "Other CI/CD provider": https://docs.dagster.io/dagster-cloud/getting-started#hybrid

**Comment by m-heinemann** [comment]
Thanks! Can I set up CI/CD using bitbucket if I'm on the Serverless deployment option?

**Reply by shalabhc** [reply]
Yes, here are the instructions for serverless: https://docs.dagster.io/dagster-cloud/deployment/serverless#other-bitbucket-or-local-development

**Reply by m-heinemann** [reply]
Thanks, I don't see anything there about CI/CD or branch deployments though

**Reply by m-heinemann** [reply]
I've been working with the docs to try to set up branch deployments via the CLI
```
BRANCH_DEPLOYMENT_NAME=$(
>     dagster-cloud branch-deployment create-or-update \
>         --organization my-org \
>         --api-token agent:my-ord:abc123\
>         --git-repo-name repo.my-org.com \
>         --branch-name mrsh \
>         --read-git-state 
```

Then

```
agster-cloud workspace add-location   
--organization my-org     
--deployment $BRANCH_DEPLOYMENT_NAME  
--api-token=agent:my-org::abc123 
--location-file ./repo.my-org.com/dagster_cloud.yaml 
--location-name quickstart_aws
--image image_url:tag
```

I keep on getting the following error:
```
Error: Some locations failed to load after being synced by the agent:
Error loading quickstart_aws: {'__typename': 'PythonError', 'message': 
"dagster._core.errors.DagsterImportError: Encountered ImportError: `No module named 'quickstart_aws'` while importing module quickstart_aws. 
Local modules were resolved using the working directory `/`. If another working directory should be used, please explicitly specify the appropriate path using the `-d` or `--working-directory` for CLI based targets or the `working_directory` configuration option for workspace targets. \n", 'stack': ['  File "/dagster/dagster/_grpc/server.py", line 295, in __init__\n    self._loaded_repositories: Optional[LoadedRepositories] = LoadedRepositories(\n', '  File "/dagster/dagster/_grpc/server.py", line 139, in __init__\n    loadable_targets = get_loadable_targets(\n', '  File "/dagster/dagster/_grpc/utils.py", line 60, in get_loadable_targets\n    else loadable_targets_from_python_package(package_name, working_directory)\n', '  File "/dagster/dagster/_core/workspace/autodiscovery.py", line 48, in loadable_targets_from_python_package\n    module = load_python_module(\n', '  File "/dagster/dagster/_core/code_pointer.py", line 140, in load_python_module\n    raise DagsterImportError(\n']}
````
Any suggestions as to what could be the issue, or tips for debugging this? Thank you!

**Reply by shalabhc** [reply]
Once you get the branch deployment name, instead of `dagster-cloud workspace add-location` I suggest using the `dagster-cloud serverless deploy-python-executable` command.

If you have a `dagster_cloud.yaml` configured, you can point to that with `dagster-cloud serverless deploy-python-executable --location-file=path/to/dagster_cloud.yaml`. You can see a sample dagster_cloud.yaml file here: https://github.com/dagster-io/dagster/tree/master/examples/quickstart_aws. In addition you should also provide `--location-name=quickstart_aws`. To specify a deployment add the `--deployment=$BRANCH_DEPLOYMENT_NAME` flag. You will need the `--organization` and `--api-token` flags as well.

**Reply by m-heinemann** [reply]
Thank you! That worked.

Maybe you could update the documentation so that others will have an easier time figuring this out:
https://docs.dagster.io/dagster-cloud/managing-deployments/branch-deployments/using-branch-deployments#prerequisites
